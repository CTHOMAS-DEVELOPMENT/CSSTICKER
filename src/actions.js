export const addPlayer = player => ({
  type: "ADD PLAYER",
  payload: player
});
export const removePlayer = playerSet => ({
  type: "REMOVE PLAYER",
  payload: playerSet
});
export const updatePlayer = (player, index) => ({
  type: "UPDATE PLAYER",
  payload: player,
  index: index
});
export const updateDice = dice => ({
  type: "UPDATE DICE",
  payload: dice
});
