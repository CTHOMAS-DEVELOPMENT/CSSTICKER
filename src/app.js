import React from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import "./styles/styles.scss";
import store from "./store";
import AppRouter from "./AppRouter";

render(
  <Provider store={store}>
    <AppRouter />
  </Provider>,
  document.getElementById("app")
);
