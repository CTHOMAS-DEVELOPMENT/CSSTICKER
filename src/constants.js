export const INITIALIZE = {
  dice: [0, 0],
  players: [
    {
      current: 0,
      total: 0,
      active: true
    },
    {
      current: 0,
      total: 0,
      active: false
    }
  ]
};
export const WINNINGSCORE = 100;
