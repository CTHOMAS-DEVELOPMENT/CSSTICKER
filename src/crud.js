import store from "./store";
import { BASETICKERURL, CURRENCYLIST } from "./constants";
import { addCryptoValues } from "./actions";
import axios from "axios";

export const cryptoCurrencyData = (bitcoin, bitcoinName) => 
{
  
  const url=`${ BASETICKERURL }fsym=${ bitcoin }&tsyms=${ CURRENCYLIST }`;
  axios.get( url )
      .then(res => {
        const cryptos= res.data;
        cryptos[bitcoin]=bitcoinName;
        store.dispatch(addCryptoValues(cryptos));
      })
      .catch(function (error) {
        // handle error
        console.log("ERROR : ",error);
      });
    }
/////
//ADD
/////
////////
//UPDATE
////////
////////
//DELETE
////////
