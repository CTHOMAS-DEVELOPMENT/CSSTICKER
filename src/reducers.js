import { INITIALIZE } from "./constants";
const initialState = INITIALIZE;
const rootReducer = (state = initialState, action) => {
  switch (action.type) {
    case "UPDATE DICE":
      return { ...state, dice: [...state.dice, action.payload] };
    case "ADD PLAYER":
      return { ...state, players: [...state.players, action.payload] };
    case "REMOVE PLAYER":
      return { ...state, players: action.payload };
    case "UPDATE PLAYER":
      return { ...state, players: [...state.players, action.payload] };
    default:
      return state;
  }
};
export default rootReducer;
