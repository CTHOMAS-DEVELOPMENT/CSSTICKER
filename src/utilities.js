//Utility functions that need no imports (Other than constants)

import { WINNINGSCORE } from "./constants";
export const getCurrentPlayer = players => {
  for (var i = 0; i < players.length; i++) {
    if (players[i].active) {
      return i;
    }
  }
  return 0;
};
export const getNextPlayer = (players, currentIndex) => {
  return currentIndex + 1 === players.length ? 0 : currentIndex + 1;
};
export const getRoleResult = (r1, r2, current, aggregate) => {
  //2 ones (Snake eyes) thrown
  if (r1 === 1 && r2 === 1) {
    return "SNAKE EYES";
  }
  //1 thrown
  if (r1 === 1 || r2 === 1) {
    return "LOSE SCORE";
  }
  //Winning score reached
  if (r1 + r2 + current + aggregate >= WINNINGSCORE) {
    return "WINNER";
  }
  return "AGGREGATE";
};
export const isStillInitialized = existingPlayers => {
  //Note: To be initialized, no player must hold any score and player 1 should be "active"
  let noScoreHeld = true;
  for (let i = 0; i < existingPlayers.length; i++) {
    if (existingPlayers[i].current !== 0 || existingPlayers[i].total !== 0) {
      noScoreHeld = false;
      break;
    }
  }

  return existingPlayers[0].active && noScoreHeld;
};
