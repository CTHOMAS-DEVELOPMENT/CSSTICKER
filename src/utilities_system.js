import { updatePlayer } from "./actions";
import { getNextPlayer } from "./utilities";

export const initializePlayerSet = players => {
  players[0].active = true;
  players[0].current = 0;
  players[0].total = 0;
  updatePlayer(players[0], 0);

  for (let i = 1; i < players.length; i++) {
    players[i].active = false;
    players[i].current = 0;
    players[i].total = 0;
    updatePlayer(players[i], 0);
  }
  return players;
};
export const setHolder = (players, index) => {
    let player = players[index];
    //Set the player to non active
    player.total += player.current;
    player.current = 0;
    player.active = false;
    //Set the next player to active
    index = getNextPlayer(players, index);
    player = players[index];
    player.active = true;
}
export const setPlayerTotalLossScore = (player, players, index) => {
  let message = `Player ${index +
    1} : You rolled two 1's and lose all your points.`;
  player.current = 0;
  player.total = 0;
  player.active = false;
  updatePlayer(player, index);
  
  //Set the active player
  index = getNextPlayer(
    players,
    index
  );
  player = players[index];
  player.active = true;
  updatePlayer(player, index);
  return message;
}
export const setPlayerLossScore = (player, players, index) => {
  let message = `Player ${index + 1} : You rolled a 1 and lose your score.`;
  player.current = 0;
  player.active = false;
  updatePlayer(player, index);

  //Set the active player
  index = getNextPlayer(players, index);
  player = players[index];
  player.active = true;
  updatePlayer(player, index);

  return message;
};
