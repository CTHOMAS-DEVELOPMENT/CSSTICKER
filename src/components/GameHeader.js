import React from "react";
import store from "../store";
import { removePlayer } from "../actions";
import { initializePlayerSet } from "../utilities_system";
import {
  faRocket,
  faPlusCircle,
  faMinusCircle
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { isStillInitialized } from "../utilities";

class GameHeader extends React.Component {
  reStart = () => {
    const players = initializePlayerSet(this.props.players);
    this.props.reStart(players);
  };
  render() {
    const classDef = "displayplayer sbtxtsz_mini";
    const classDefActive = "displayplayer sbtxtsz_mini live";
    return (
      <div>
        <div className="panelitem headertop">
          {this.props.players.length > 2 &&
            isStillInitialized(this.props.players) && (
              <button
                href=""
                className="btn"
                onClick={() => {
                  let players = this.props.players.slice(0, -1);
                  store.dispatch(removePlayer(players));
                  this.props.removePlayer();
                }}
                title="Remove a player!"
              >
                <FontAwesomeIcon icon={faMinusCircle} className="svgicon" />
              </button>
            )}

          <button
            href=""
            className="btn"
            onClick={() => {
              this.reStart();
            }}
            title="Start again!"
          >
            <FontAwesomeIcon icon={faRocket} className="svgicon" />
          </button>

          {isStillInitialized(this.props.players) && (
            <button
              href=""
              className="btn"
              onClick={() => {
                this.props.addPlayer();
              }}
              title="Add a player!"
            >
              <FontAwesomeIcon icon={faPlusCircle} className="svgicon" />
            </button>
          )}
        </div>
        <div className="panelitem">
          {this.props.players.map((player, i) => (
            <span className={player.active ? classDefActive : classDef} key={i}>
              Player {i + 1} [{player.current}] [{player.total}]
            </span>
          ))}
        </div>
      </div>
    );
  }
}

export default GameHeader;
