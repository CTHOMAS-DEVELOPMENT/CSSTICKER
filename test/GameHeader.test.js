import React from "react";
import { shallow } from "enzyme";
import GameHeader from "../src/components/GameHeader";
import { INITIALIZE } from "../src/constants";

test("Should render GameHeader correctly..", () => {
  const wrapper = shallow(<GameHeader players={INITIALIZE.players} />);
  expect(wrapper).toMatchSnapshot();

});
