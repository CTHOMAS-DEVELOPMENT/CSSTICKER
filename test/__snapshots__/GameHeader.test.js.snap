// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render GameHeader correctly.. 1`] = `
<div>
  <div
    className="panelitem headertop"
  >
    <button
      className="btn"
      href=""
      onClick={[Function]}
      title="Start again!"
    >
      <FontAwesomeIcon
        border={false}
        className="svgicon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f135",
              "M505.1 19.1C503.8 13 499 8.2 492.9 6.9 460.7 0 435.5 0 410.4 0 307.2 0 245.3 55.2 199.1 128H94.9c-18.2 0-34.8 10.3-42.9 26.5L2.6 253.3c-8 16 3.6 34.7 21.5 34.7h95.1c-5.9 12.8-11.9 25.5-18 37.7-3.1 6.2-1.9 13.6 3 18.5l63.6 63.6c4.9 4.9 12.3 6.1 18.5 3 12.2-6.1 24.9-12 37.7-17.9V488c0 17.8 18.8 29.4 34.7 21.5l98.7-49.4c16.3-8.1 26.5-24.8 26.5-42.9V312.8c72.6-46.3 128-108.4 128-211.1.1-25.2.1-50.4-6.8-82.6zM400 160c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z",
            ],
            "iconName": "rocket",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <button
      className="btn"
      href=""
      onClick={[Function]}
      title="Add a player!"
    >
      <FontAwesomeIcon
        border={false}
        className="svgicon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f055",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
            ],
            "iconName": "plus-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </div>
  <div
    className="panelitem"
  >
    <span
      className="displayplayer sbtxtsz_mini live"
      key="0"
    >
      Player 
      1
       [
      0
      ] [
      0
      ]
    </span>
    <span
      className="displayplayer sbtxtsz_mini"
      key="1"
    >
      Player 
      2
       [
      0
      ] [
      0
      ]
    </span>
  </div>
</div>
`;
